core/src/main/scala/com/microsoft/azure/synapse/ml/causal/BaseDiffInDiffEstimator.scala

```diff
@@ -0,0 +1,164 @@
+// Copyright (C) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in project root for information.
+
+package com.microsoft.azure.synapse.ml.causal
+
+import com.microsoft.azure.synapse.ml.causal.linalg.DVector
+import com.microsoft.azure.synapse.ml.codegen.Wrappable
+import com.microsoft.azure.synapse.ml.logging.{FeatureNames, SynapseMLLogging}
+import com.microsoft.azure.synapse.ml.param.DataFrameParam
+import org.apache.spark.SparkException
+import org.apache.spark.ml.feature.VectorAssembler
+import org.apache.spark.ml.param.ParamMap
+import org.apache.spark.ml.regression.LinearRegression
+import org.apache.spark.ml.util.Identifiable
+import org.apache.spark.ml.{ComplexParamsReadable, ComplexParamsWritable, Estimator, Model}
+import org.apache.spark.sql.types.{BooleanType, NumericType, StructField, StructType}
+import org.apache.spark.sql.{DataFrame, Dataset}
+
+import java.util
+
+abstract class BaseDiffInDiffEstimator(override val uid: String)
+  extends Estimator[DiffInDiffModel]
+    with DiffInDiffEstimatorParams {
+
+  private def validateFieldNumericOrBooleanType(field: StructField): Unit = {
+    val dataType = field.dataType
+    require(dataType.isInstanceOf[NumericType] || dataType == BooleanType,
+      s"Column ${field.name} must be numeric type or boolean type, but got $dataType instead.")
+  }
+
+  protected def validateFieldNumericType(field: StructField): Unit = {
+    val dataType = field.dataType
+    require(dataType.isInstanceOf[NumericType],
+      s"Column ${field.name} must be numeric type, but got $dataType instead.")
+  }
+
+  override def transformSchema(schema: StructType): StructType = {
+    validateFieldNumericOrBooleanType(schema(getPostTreatmentCol))
+    validateFieldNumericOrBooleanType(schema(getTreatmentCol))
+    validateFieldNumericType(schema(getOutcomeCol))
+    schema
+  }
+
+  override def copy(extra: ParamMap): Estimator[DiffInDiffModel] = defaultCopy(extra)
+
+  private[causal] val interactionCol = "interaction"
```

Comment:
Is this a column thats used in an output? If so should be settable. If its something intermediate use the findUnusedColumn method to ensure no clashes

Reply:
it's an intermediate column - but yes findUnusedColumn is better. I'll update the code.

Reply:
done.